doc/design/diagrams/iceoryx_components_diagram_v3_0_0.puml

```diff
@@ -27,26 +28,42 @@ skinparam frame {
 
 header Eclipse iceoryx component overview
 
+' Platform Library containing platform abstraction with quality level 1
+package iceoryx_platform  {
+   frame "platform library" {
+      component "Platform abstraction" <<module>>
+   }
+}
+
 ' HOOFS: Helpful Objects Optimised For Safety
-' Library containing modern C++ STL constructs
+' Library containing modern C++ STL constructs with quality level 1+
 package iceoryx_hoofs  {
    frame "hoofs library" {
-      component concurrent <<namespace>>
-      component cxx <<namespace>> {
-         component newtype <<namespace>>
-         component set <<namespace>>
-         component algorithm <<namespace>>
-      }
-      component DesignPattern <<namespace>>
-      component errors <<namespace>>
-      component log <<namespace>> {
-         component ffbb <<namespace>>
-      }
-      component units <<namespace>> {
-         component duration_literals <<namespace>>
-
-      }
-      component posix <<namespace>>
+
+      component "Memory & lifetime management" <<module>>
+      component Containers <<module>>
+      component "Common vocabulary types" <<module>>
```

Comment:
When there are common vocabulary types are there also uncommon or special vocabulary types? I would follow YAGNI and say just vocabulary types (less to type and more concise).

Reply:
@elfenpiff I find [this explanation](https://www.oreilly.com/library/view/mastering-the-c17/9781787126824/c6737dd3-47fa-4719-a459-72afac90857c.xhtml) from  quite plausible: `A "vocabulary" type is a type that purports to provide a single lingua franca, a common language, for dealing with its domain.` AFAIK C++17 called `optional` and `variant` common vocabulary types. I would e.g. say that string might be seen by some as a special/uncommon one. So yes you are right, let's simplify this.